csrc/xdh_gen.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+#include "auto_free.h"
+#include "env.h"
+#include "generated-headers.h"
+#include <openssl/evp.h>
+
+using namespace AmazonCorrettoCryptoProvider;
+
+static void generateXECKey(EVP_PKEY_auto& key)
+{
+    EVP_PKEY_CTX_auto ctx = EVP_PKEY_CTX_auto::from(EVP_PKEY_CTX_new_id(EVP_PKEY_X25519, nullptr));
```

Comment:
There's nothing in this file that's specific to x25519 other than the `EVP_PKEY_X25519` constant. You can refactor this file and `ed_gen.cpp` to accept this constant from the java layer to remove the code repetition. This also leaves the door open to a future x448 implementation and other algorithms that use this interface.